From 00235620c43682ea3500cc5a7a89ff328b6f8625 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Thu, 7 Jan 2021 12:00:57 +0100
Subject: [PATCH] Makefile: add TARGET option

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 442ad68..0eaca77 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,9 @@ PREFIX ?= /usr
 INSTALL_DIR ?= $(DESTDIR)$(PREFIX)/lib/lv2
 CXXFLAGS ?= -g -O3
 CXX ?= g++
+TARGET ?= gate_gui.so abGateQt/libabGateQt.so gate.so
 
-$(BUNDLE): manifest.ttl gate.ttl gate.so gate_gui.so bypass_on.png bypass_off.png knob.png background.png abGateQt/libabGateQt.so
+$(BUNDLE): manifest.ttl gate.ttl bypass_on.png bypass_off.png knob.png background.png $(TARGET)
 	rm -rf $(BUNDLE)
 	mkdir $(BUNDLE)
 	cp $^ $(BUNDLE)
