From 69c6b01d0cffecfcf1055111a37f12ec80617317 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Thu, 15 Jun 2017 22:32:47 +0200
Subject: [PATCH] build: disable object stripping

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 871f6ee..ae9eb1c 100644
--- a/Makefile
+++ b/Makefile
@@ -250,7 +250,6 @@ $(BUILDDIR)$(LV2NAME)$(LIB_EXT): $(DSP_DEPS) Makefile
 	$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(LIC_CFLAGS)\
 	  -o $(BUILDDIR)$(LV2NAME)$(LIB_EXT) $(DSP_SRC) \
 	  -shared $(LV2LDFLAGS) $(LDFLAGS) $(LOADLIBES) $(LIC_LOADLIBES)
-	$(STRIP) $(STRIPFLAGS) $(BUILDDIR)$(LV2NAME)$(LIB_EXT)
 
 jackapps: $(JACKAPP)
 
-- 
1.9.1

