From 3708d5e9d441f1b2f85970c36aec12893f6cecd6 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Sat, 17 Oct 2020 15:03:53 +0200
Subject: [PATCH] webserver: do not wait for cc

---
 mod/webserver.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/mod/webserver.py b/mod/webserver.py
index c1ff632..1f00a13 100644
--- a/mod/webserver.py
+++ b/mod/webserver.py
@@ -1376,9 +1376,6 @@ class TemplateHandler(TimelessRequestHandler):
         loader = Loader(HTML_DIR)
         section = path.split('.',1)[0]
 
-        if section == 'index':
-            yield gen.Task(SESSION.wait_for_hardware_if_needed)
-
         try:
             context = getattr(self, section)()
         except AttributeError:
