From 4e73930815c511ee4af04fdbf40d92c59190e618 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Wed, 28 Jun 2017 11:47:15 +0200
Subject: [PATCH] build: clean LDFLAGS variable

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 383a33d..3e25d13 100644
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,6 @@ HEADERS = $(addsuffix /plugin.h, $(BUILD_DIRS))
 PLUGINS = $(addsuffix /pdlv2.so, $(BUILD_DIRS))
 
 LVTKLIB = lvtk/build/src/liblvtk_plugin2.a
-LDFLAGS += -ldl ${LVTKLIB}
 CXXFLAGS += -Wl,--no-as-needed -Wno-narrowing -shared -fPIC -DPIC -Isrc/ -std=c++11 -Ilvtk/
 
 LIBPD_FLAGS = UTIL=true EXTRA=true
@@ -22,7 +21,7 @@ RUBY = ruby
 all: $(PLUGINS)
 
 $(BUILD_DIR)/pdlv2-%.lv2/pdlv2.so: $(BUILD_DIR)/pdlv2-%.lv2/plugin.h src/plugin.cpp $(LVTKLIB)
-	$(CXX) $(CXXFLAGS) src/plugin.cpp -I$(dir $<) -o $@ $(LDFLAGS)
+	$(CXX) $(CXXFLAGS) src/plugin.cpp -I$(dir $<) -Iinclude -o $@ $(LDFLAGS) -ldl ${LVTKLIB}
 
 $(BUILD_DIR)/pdlv2-%.lv2/plugin.h: plugins/%/plugin.pd src/process.rb src/host.pd $(LIBPD_SO)
 	$(RUBY) src/process.rb $< $(dir $@)
