From aed7a1e375cab21aef341dbc846c521e826eecd6 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Sat, 16 Sep 2017 11:45:15 +0200
Subject: [PATCH 2/2] build: install modgui subdir

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c269ba..416106c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,5 +100,5 @@ set_target_properties(artyfx PROPERTIES PREFIX "")
 # add the install targets
 install (TARGETS artyfx DESTINATION lib/lv2/artyfx.lv2/)
 
-FILE(GLOB ttls "artyfx.lv2/*")
+FILE(GLOB_RECURSE ttls "artyfx.lv2/*")
 install(FILES ${ttls} DESTINATION lib/lv2/artyfx.lv2/)
-- 
1.9.1

