From e3cdcb4215022cae503d4d12c527f478861010ae Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Sat, 15 Jul 2017 14:01:35 +0200
Subject: [PATCH] build: disable TTL generation

---
 Makefile | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Makefile b/Makefile
index c82c7e6..eefd0b0 100644
--- a/Makefile
+++ b/Makefile
@@ -69,12 +69,8 @@ gverb: gverb/gverb.c gverb/gverbdsp.c gverb/gverb.o gverb/gverbdsp.o
 util: util/blo.o util/iir.o util/db.o util/rms.o util/pitchscale.o
 
 %.c: OBJ = $(shell echo $@ | sed 's/\.c$$/-@OS@.$(EXT)/')
-%.c: %.xml xslt/source.xsl xslt/manifest.xsl
-	xsltproc -novalid xslt/source.xsl $*.xml | sed 's/LADSPA_Data/float/g' > $@
-	xsltproc -novalid -stringparam obj `basename $(OBJ)` xslt/manifest.xsl $*.xml > `dirname $@`/manifest.ttl.in
 
 %.ttl: %.xml xslt/turtle.xsl
-	xsltproc -novalid xslt/turtle.xsl $*.xml | sed 's/\\/\\\\/g' > $@
 
 %.o: NAME = $(shell echo -n $@ | sed 's/plugins\/\(.*\)-swh.*/\1/')
 %.o: %.c
@@ -84,7 +80,6 @@ util: util/blo.o util/iir.o util/db.o util/rms.o util/pitchscale.o
 %.$(EXT): %.xml %.o %.ttl
 	$(CC) $*.o $(PLUGIN_LDFLAGS) $($(NAME)_LDFLAGS) -o $@
 	cp $@ $*-$(OS).$(EXT)
-	sed 's/@OS@/$(OS)/g' < `dirname $@`/manifest.ttl.in > `dirname $@`/manifest.ttl
 
 clean: dist-clean
 
-- 
1.9.1

