From 538364bc91b7b44f8f93ef82133024917d3a4875 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Sun, 20 Aug 2017 10:01:19 +0200
Subject: [PATCH 1/2] calfmakerdf: allow build in cross environment

---
 src/Makefile.am | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 794e668..9bf0ab8 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -31,8 +31,15 @@ endif
 
 AM_CXXFLAGS += $(GLIB_DEPS_CFLAGS)
 noinst_PROGRAMS += calfmakerdf
-calfmakerdf_SOURCES = makerdf.cpp
-calfmakerdf_LDADD = calf.la
+calfmakerdf_SOURCES = makerdf.cpp giface.cpp preset.cpp utils.cpp metadata.cpp
+calfmakerdf_LDADD = -lexpat
+calfmakerdf_LD = $(CXX_FOR_BUILD)
+calfmakerdf_LDFLAGS = $(LDFLAGS_FOR_BUILD)
+calfmakerdf_CXXFLAGS = $(CXXFLAGS_FOR_BUILD)
+calfmakerdf_LINK = $(calfmakerdf_LD) $(calfmakerdf_LDFLAGS) -o $@
+
+$(calfmakerdf_OBJECTS): CXX=$(CXX_FOR_BUILD)
+$(calfmakerdf_OBJECTS): CXXFLAGS=$(CXXFLAGS_FOR_BUILD)
 
 calfbenchmark_SOURCES = benchmark.cpp
 calfbenchmark_LDADD = calf.la
-- 
2.7.4

