From c635984339a7ecad083e6349b140c1aa8e2254d0 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Thu, 15 Jun 2017 21:48:36 +0200
Subject: [PATCH] build: remove object stripping and O3 optimization

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 543ca46..d0d182d 100644
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@ ifeq ($(DEBUG), 1)
    LDFLAGS +=
 else
    CFLAGS += -fvisibility=hidden
-   LDFLAGS += -s
+   LDFLAGS += 
 endif
 
 ifeq ($(TESTBUILD), 1)
