From 41441c207260765042c5100f59cfcf44c580d456 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Thu, 15 Jun 2017 21:48:36 +0200
Subject: [PATCH] build: remove object stripping and O3 optimization

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 80833d9..16fa35c 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ SHAREDIR = $(PREFIX)/share
 MANDIR = $(SHAREDIR)/man/man1/
 
 # default compiler and linker flags
-CFLAGS += -O3 -Wall -Wextra -c -std=gnu99 -fPIC -D_GNU_SOURCE
+CFLAGS += -Wall -Wextra -c -std=gnu99 -fPIC -D_GNU_SOURCE
 LDFLAGS += -Wl,--no-undefined
 
 # debug mode compiler and linker flags
@@ -26,7 +26,7 @@ ifeq ($(DEBUG), 1)
    LDFLAGS +=
 else
    CFLAGS += -fvisibility=hidden
-   LDFLAGS += -s
+   LDFLAGS += 
 endif
 
 ifeq ($(TESTBUILD), 1)
-- 
1.9.1

