From c8a7e2844008e72c895e9b7277c02f44178398f8 Mon Sep 17 00:00:00 2001
From: auto3000 <auto3000@users.noreply.github.com>
Date: Sun, 15 Oct 2017 23:13:11 +0200
Subject: [PATCH 4/7] mutex: disable build flags inheritance protocol

---
 src/effects.c             | 2 --
 src/rtmempool/rtmempool.c | 2 --
 2 files changed, 4 deletions(-)

diff --git a/src/effects.c b/src/effects.c
index ae2aa17..f4b5bcd 100644
--- a/src/effects.c
+++ b/src/effects.c
@@ -2086,9 +2086,7 @@ int effects_init(void* client)
 
     pthread_mutexattr_t atts;
     pthread_mutexattr_init(&atts);
-#ifdef __MOD_DEVICES__
     pthread_mutexattr_setprotocol(&atts, PTHREAD_PRIO_INHERIT);
-#endif
 
     pthread_mutex_init(&g_rtsafe_mutex, &atts);
     pthread_mutex_init(&g_midi_learning_mutex, &atts);
diff --git a/src/rtmempool/rtmempool.c b/src/rtmempool/rtmempool.c
index aef9b70..f7215e0 100644
--- a/src/rtmempool/rtmempool.c
+++ b/src/rtmempool/rtmempool.c
@@ -87,9 +87,7 @@ bool rtsafe_memory_pool_create(RtMemPool_Handle* handlePtr,
 
     pthread_mutexattr_t atts;
     pthread_mutexattr_init(&atts);
-#ifdef __ARM_ARCH_7A__
     pthread_mutexattr_setprotocol(&atts, PTHREAD_PRIO_INHERIT);
-#endif
     pthread_mutex_init(&poolPtr->mutex, &atts);
     pthread_mutexattr_destroy(&atts);
 
-- 
1.9.1

